src/db/state/record.rs

```diff
@@ -98,6 +98,29 @@ impl<'conn> State<'conn, RecordRow> {
         Ok(())
     }
 
+    /// Change the canonical id of the row.
+    ///
+    /// Returns `false` if the new canonical id already exists, and `true` otherwise.
+    pub fn change_row_canonical_id(&self, new_id: &RemoteId) -> Result<bool, rusqlite::Error> {
```

Comment:
Probably this should just be `change_canonical_id`? Since it is a method on `row` already